### ISSUE TITLE
About finding negative markers

### ISSUE THREAD

**Issue by QingShan-Aspie** [elolab/ILoReg#3]
Hi,

I was trying it out and I wonder if there's an option of getting negative markers as well?

**Comment by j0ntt** [comment]
Hi,

By that I assume you mean markers with negative fold-change? If that's the case, then 

`top10_log2FC <- SelectTopGenes(gene_markers,
                               top.N = 10,
                               criterion.type = "log2FC",
                               reverse = TRUE)`

should give you the markers with the lowest average fold-change. Additionally, 

`top10_adj.p.value <- SelectTopGenes(gene_markers,
                                    top.N = 10,
                                    criterion.type = "adj.p.value",
                                    reverse = TRUE)`

gives you the markers with the lowest adjusted p-values, regardless of the sign of their fold-change values.

I hope this helps.

**Comment by j0ntt** [comment]
Also, I'm planning to update the documentation soon. I would consider increasing the L value from 30 to say 100. For the toy dataset in the example 30 is sufficient.

**Comment by QingshanNeuro** [comment]
Thank you for your answer Johannes. I wonder if there's a criterion of how to choose an L value for different datasets?



At 2020-01-28 16:08:26, "Johannes Smolander" <notifications@github.com> wrote:

Also, I'm planning to update the documentation soon. I would consider increasing the L value from 30 to say 100. For the toy dataset in the example 30 is sufficient.

**Comment by j0ntt** [comment]
`Thank you for your answer Johannes. I wonder if there's a criterion of how to choose an L value for different datasets?` 

Hi QingShan, 

The number of ICP runs (L) is similar to the number of k-means runs parameter in the SC3 clustering algorithm. Both ILoReg and SC3 are consensus clustering methods, i.e. they combine a set of subclusterings to obtain a more accurate clustering. The designers of SC3 set the default to 1000 k-means runs, but found that with 50 k-means runs the results were only a little worse. We came to the same conclusion when we tested this using a dataset that has 3,000 cells. A higher L value would be better, but increasing L increases the run time linearly. We designed the toy dataset example to be fast, so that people can a feeling of the pipeline without having to wait long, whis is also a requirement if we are to put this to Bioconductor. 

With that said, the answer to your question about whether there is some criterion of how to choose L: no, but a higher L is preferable. I would say it depends on the size of the dataset. With a larger dataset it is possible to identify more cell types, so a higher L is likely to be needed.

P.S. I should probably now add another example that has at least 3,000 cells and use a higher L value, as this is how ILoReg should be generally used.
